group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorRuntimeMetrics.java

```diff
@@ -202,6 +248,19 @@ public void recordEventQueueTime(long durationMs) { }
     @Override
     public void recordEventQueueProcessingTime(long durationMs) { }
 
+    @Override
+    public void recordEventPurgatoryTime(long purgatoryTimeMs, long writeTimestamp) {
+        if (writeTimestamp != NOT_WRITTEN) {
+            // Only record the purgatory time if records were actually appended to the local log.
+            eventPurgatoryTimeSensor.record(purgatoryTimeMs);
+        }
```

Comment:
Ah, I see why you pass `writeTimestamp`. I wonder if we rather let the caller take this decision. Otherwise, other implementation of the interface maybe inconsistent. What do you think?